lib/dry/transaction/step_failure.rb

```diff
@@ -12,6 +12,10 @@ def method_missing(name, *args, &block)
         @__object.send(name, *args, &block)
       end
 
+      def respond_to_missing?(*args)
+        @__object.respond_to?(*args)
```

Comment:
Good catch about default values. Should we add `|| super`? I mean there is no `#respond_to_missing?` at BasicObject so that call would raise a weird error.